### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by liuda-chuzdiuk**

### CODE REVIEW

**Review by alix-mak — APPROVED** [review]

File: tests/test_main.py [hunk 1]
```diff
@@ -1 +1,27 @@
-# write your code here
+from unittest import mock
+
+from app.main import cryptocurrency_action
+
+
+@mock.patch("app.main.get_exchange_rate_prediction")
+def test_prediction_is_not_substantive(mocked_prediction):
+    mocked_prediction.return_value = 3
```
Comment (MasterpieceElbow) [inline tests/test_main.py:8]: Add some boundary conditions, what if mocked.returned value = 1,05 * value passed in `cryptocurrency_action`?

File: tests/test_main.py [hunk 2]
```diff
@@ -1 +1,27 @@
...
+
+    assert cryptocurrency_action(3) == "Do nothing"
+    assert cryptocurrency_action(3.1546) == "Do nothing"
+    assert cryptocurrency_action(2.8598) == "Do nothing"
```
Comment (MasterpieceElbow) [inline tests/test_main.py:12]: Add boundary conditions, you don't check when `get_exchange_rate_prediction` return value equals exactly 1.05 or 0.95 * value passed in `cryptocurrency_action`.

**Review by MasterpieceElbow — CHANGES_REQUESTED** [review]

File: tests/test_main.py [hunk 3]
```diff
@@ -1 +1,32 @@
...
+from unittest import mock
+
+from app.main import cryptocurrency_action
+
+
+@mock.patch("app.main.get_exchange_rate_prediction")
+def test_prediction_is_higher_than_5(mocked_prediction):
+    mocked_prediction.return_value = 1.0
+
+    assert cryptocurrency_action(1.05) == "Do nothing"
```
Comment (MasterpieceElbow) [inline tests/test_main.py:10]: `1` = 0.95238... * 1.05, you already have checked when predicted value equals exactly 0.95 of initial value below but you still haven't check boundary values when prediction equals **exactly** 1.05 of initial value.

**Review by MasterpieceElbow — CHANGES_REQUESTED** [review]

File: tests/test_main.py (hunk 3 above)
Comment (liuda-chuzdiuk) [reply tests/test_main.py]: fixed

File: tests/test_main.py [hunk 4]
```diff
@@ -1 +1,31 @@
-# write your code here
+from unittest import mock
+
+from app.main import cryptocurrency_action
+
+
+@mock.patch("app.main.get_exchange_rate_prediction")
+def test_prediction_is_higher_exactly_5(mocked_prediction):
```
Comment (IvanRamyk) [inline tests/test_main.py:7]:
```suggestion
def test_prediction_is_higher_exactly_by_5_percents(mocked_prediction):
```

File: tests/test_main.py [hunk 5]
```diff
@@ -1 +1,31 @@
...
+    mocked_prediction.return_value = 1.05
+
+    assert cryptocurrency_action(1.0) == "Do nothing"
+
+
+@mock.patch("app.main.get_exchange_rate_prediction")
+def test_prediction_is_lower_exactly_5(mocked_prediction):
```
Comment (IvanRamyk) [inline tests/test_main.py:14]:
```suggestion
def test_prediction_is_lower_exactly_by_5_percents(mocked_prediction):
```

**Review by IvanRamyk — CHANGES_REQUESTED** [review]

**Review by IvanRamyk — APPROVED** [review]
Well done!
